src/vma/lwip/tcp_out.c

```diff
@@ -422,7 +422,7 @@ tcp_write(struct tcp_pcb *pcb, const void *arg, u32_t len, u8_t apiflags)
     /* Usable space at the end of the last unsent segment */
     unsent_optlen = LWIP_TCP_OPT_LENGTH(pcb->last_unsent->flags);
     LWIP_ASSERT("mss_local is too small", mss_local >= pcb->last_unsent->len + unsent_optlen);
-    space = mss_local - (pcb->last_unsent->len + unsent_optlen);
+    space =  ((mss_local > (pcb->last_unsent->len + unsent_optlen)) ? (mss_local - (pcb->last_unsent->len + unsent_optlen)) : 0);
```

Comment:
Please reverse expression. 
It's easier to read if the short part ('0') comes first and then the long expression.